plutus-core/plutus-core/src/PlutusCore/MkPlc.hs

```diff
@@ -317,3 +321,16 @@ mkIterKindArrow
     -> Kind ann
     -> Kind ann
 mkIterKindArrow ann kinds target = foldr (KindArrow ann) target kinds
+
+{- | A helper to create a single, fresh strict binding; It returns the fresh bound `Var`iable and
+a function `Term -> Term`, expecting an "in-Term" to form a let-expression.
+-}
+mkFreshTermLet :: (MonadQuote m, TermLike t tyname Name uni fun, Monoid a)
+               => Type tyname uni a -- ^ the type of binding
+               -> t a -- ^ the term bound to the fresh variable
+               -> m (t a, t a -> t a) -- ^ the fresh Var and a function that takes an "in" term to construct the Let
+mkFreshTermLet aT a = do
+    -- I wish this was less constrained to Name
+    genName <- freshName "generated"
```

Comment:
Overall I think it's worth making the caller pass the name string. It's quite helpful having a name that says _something_ about why it was generated.